ranger-zookeeper/src/main/java/com/flipkart/ranger/zookeeper/servicefinder/ZkNodeDataSource.java

```diff
@@ -71,6 +88,7 @@ public ZkNodeDataSource(
         }
         catch (Exception e) {
             log.error("Error getting service data from zookeeper: ", e);
+            Exceptions.illegalState(e);
```

Comment:
This might not be an illegal state na? Could be a zk conn timeout.

Reply:
True. the readChild in the method is what throws an exception, in the readChild block the NoNodeException is already caught and logged for, we may not need to throw one more from here.

Reply:
Addressed it [here](https://github.com/flipkart-incubator/ranger/pull/44/commits/fb694f58ba80ad643d55d4593571b271eb9c461f)